src/java/main/com/weblyzard/api/serialize/json/DocumentHeaderDeserializer.java

```diff
@@ -18,6 +21,30 @@
 
     @Override
     public Object deserializeKey(String key, DeserializationContext ctxt) throws IOException {
+
+        // check if key represents a supported full qualified URI
+        Optional<Namespace> optionalNamespace = Arrays.stream(Namespace.values())
+                .filter(namespace -> key.startsWith(namespace.getNamespace())).findFirst();
+
```

Comment:
is `startsWith` robust enough?
(i.e. what happens if we have `xs`, `xsl` and `xslt` prefixes.)

Reply:
this part checks if key represents a full qualified uri that we support, as in https://github.com/weblyzard/weblyzard_api/blob/b340025fc2094be96cfbedeb3783ed9d51fa5477/src/java/test/com/weblyzard/api/serialize/json/DocumentHeaderDeserializerTest.java#L22-L29

since all strings in the search index end in a separator-character, a startsWith is enough.